client/src/pages/SignUp.js

```diff
@@ -13,19 +14,40 @@ import LockOutlinedIcon from '@mui/icons-material/LockOutlined';
 import Typography from '@mui/material/Typography';
 import Container from '@mui/material/Container';
 import { createTheme, ThemeProvider } from '@mui/material/styles';
+import { useNavigate } from "react-router-dom";
 
 const theme = createTheme();
 
-export default function SignUp() {
-  const handleSubmit = (event) => {
-    event.preventDefault();
-    const data = new FormData(event.currentTarget);
-    console.log({
-      email: data.get('email'),
-      password: data.get('password'),
+const SignUp= () => {
+  const navigate = useNavigate();
+  const [email, setEmail] = React.useState("");
+  const [password, setPassword] = React.useState("");
+  const [firstName, setFirstName] = React.useState("");
+  const [lastName, setLastName] = React.useState("");
+
+
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    Axios.post('http://localhost:3300/api/user/register', {
+      email,
+      password,
+      firstName,
+      lastName
+    })
+    .then((res) => {
+      console.log("sending request",res);
+    })
+    .catch((err) => {
+      console.log(err);
     });
+    navigate("/signin");
```

Comment:
I think we should move this line into the `then((res) => ...)` clause of the post call. The reason being that navigate in it's current position will be executed before we know the result from the post request. This way we will only move to the signin screen after we know that call has been successful